lib/internal/test_runner/test.js

```diff
@@ -575,12 +569,20 @@ class Test extends AsyncResource {
         return;
       }
 
-      await afterEach();
-      await after();
+      if (this.parent?.hooks.afterEach.length > 0) {
+        await afterEach();
+      }
+      if (this.hooks.after.length > 0) {
+        await this.runHook('after', { args, ctx });
+      }
       this.pass();
     } catch (err) {
-      try { await after(); } catch { /* Ignore error. */ }
-      try { await afterEach(); } catch { /* test is already failing, let's ignore the error */ }
+      if (this.parent?.hooks.afterEach.length > 0) {
+        try { await afterEach(); } catch { /* test is already failing, let's ignore the error */ }
+      }
+      if (this.hooks.after.length > 0) {
+        try { await this.runHook('after', { args, ctx }); } catch { /* Ignore error. */ }
+      }
```

Comment:
@MoLow is the change of order of hooks intentional? (`after > afterEach` -> `afterEach > after`)
I see prior to this change they are ordered differently in `try` than in `catch` 🤔

Reply:
yes, I changed the order to match what happens when test passes